### DISCUSSION TITLE
Connection stopped working

### DISCUSSION THREAD

**Discussion by sadiq81** [wlcrs/huawei_solar#786] [Connecting to the inverter]
Hi

I know this is the most common issue with this integration but I was hopping someone could point me to the problem

My hardware setup is the Home assistant Yellow. I’m connected to the dongle via WiFi and the integration has been working for several weeks without issues. Now it won’t connect. I’ve attached some SS of the logs and my setup. 

![IMG_4194](https://github.com/user-attachments/assets/ce491331-1ef4-46a1-a53e-16ba9437b7b2)
![IMG_4195](https://github.com/user-attachments/assets/b8b8777d-8abf-4275-9e5c-ed7d1f47e5f8)
![IMG_4196](https://github.com/user-attachments/assets/c71380ef-fca4-4856-919e-4d098e593158)
![IMG_4199](https://github.com/user-attachments/assets/00b58bbf-2007-4575-85e5-b0c2f0eadbbf)
![IMG_4200](https://github.com/user-attachments/assets/44ce65f3-45b9-49ca-9a78-99d444294c6b)

**Comment by Roving-Ronin** [comment]
Check to make sure your sDongle has 'unlimited' set for the IP address range(s) that can connect to it, not a specific single IP address.
Also I'd be changing your HA instance to use a fixed IP address, not a DHCP assigned IP address. Go to Settings -->  System --> Network and click on the IPv4 in 'configure network interfaces' and change to 'Static' and enter the fixed IP settings to use.
Whilst in there, I'd also be disabling IPv6.

Your Inverter could do with an update of firmware to SPC160 (there are newer, but some regions haven't been 'allowed' to see and use it by Huawei support). The sDongle v2xxxSPC108 is OK, but you could also update to v2xxxSPC110. I'm running both of these versions and it is rock solid with no drop outs (even on WiFi) for months (6+ since last updated). If you have disconnects, then start looking at your home network setup, especially if using WiFi.

https://github.com/wlcrs/huawei_solar/wiki/Confirmed-firmware-versions#three-phase---sun2000-m1-hybrid

**Comment by tparvais** [comment]
same issue, not reporting anymore since yesterday despite it worked fine during several weeks. yesterday I upgraded HA (running on green) to last release  
Core 2024.8.2
Supervisor 2024.08.0
Operating System 13.0
Interface utilisateur 20240809.0

my install: 
Dongle FW: V200R022C10SPC114
Inverter FW : V200R001C00SPC141

inverter SUN2000-2KTL-L1
dongle SDongleA-05
connected directly via ethernet

fusionsolar is still working fine

**Reply by tparvais** [reply]
I'm doing upgrade to last FW available 
V200R022C10SPC118 for dongle
V200R001C00SPC142 |  for inverter

let's see...

**Reply by tparvais** [reply]
don't know if this FW upgade solves the issue but it works again...

**Comment by se4n01** [comment]
I have a similar issue, I thought I'd use this topic. Basically sometimes the usb wifi is momentarily disconnected for some reason and dmesg reports it is unplugged and plugged back in. I've tried varios ports but it happens quite regularly. When this happens homeassistant does not redetect the usb and reconnect to the network, I have to restart homeassistant (very ugly). 

I'm not sure its even an issue with this project, more likely it is a firmware or kernel issue or a homeassistant bug that its not restoring the net. Anyway, it would be nice if there was consensus on how to better handle this, around once a week i am rebooting my homeassistant server for this problem which is not ideal.

**Reply by se4n01** [reply]
after a bit of poking around in the usb subsystem I found this:
```
cat /sys/module/usbcore/parameters/autosuspend
2
```
I have disabled autosuspend, will report back in a few days if the issue is gone.
